### PR TITLE
Validations

### PR DESCRIPTION
**Pull request by Mahdi-Niazi**
### In this milestone I:

- Add the following validations:
- For the User model:
- Name must not be blank.
- PostsCounter must be an integer greater than or equal to zero.
- For the Post model:
- Title must not be blank.
- Title must not exceed 250 characters.
- CommentsCounter must be an integer greater than or equal to zero.
- LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  end
+
+  it 'validates that likes_counter is greater than or equal to 0' do
+    post.likes_counter = -1
+    expect(post).to_not be_valid
+  end
+
+  it 'returns the five most recent comments for the post' do
+    expect(post.recent_comments).to eq(post.comments.order(created_at: :desc).limit(5))
+  end
```
Comment (Shedrack-Sunday) [inline spec/post_spec.rb:37]: - [ ] Based on project requirements, you're expected to Add unit specs for all of your models' methods and validations.
- You did this in the `user` test, which i must commend you for 👏 . Kindly do same for the other three models.

#
## Sample as done in user model test:
```
describe 'validations' do
  it { should validate_presence_of(:name) }
end
```
### Requirement:
image

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @Mahdi-Niazi  👋 

# CHANGES REQUIRED ♻️

Impressive work on the project so far👏
You've done extremely well so far in this project,  However, there are some adjustments you need to make to move to the next project. 💯



## Highlights   
- Good use of GitHub features ✔️   
- Nice work with Rails model✔️
- Kudus creating a couple pf model validations      ✔️  

# - Required Changes  🌵  

- [ ] Kindly implement model validations for all  files as you did in  `user`  model test

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing really improves your application.



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by Mahdi-Niazi** [comment]
Dear Reviewer,

I am currently in terminal have a flight I have done this changes kindly approved and please ignore if it have any error cause I don't have time to correct. 


Thank you in advance!

**Review by joel-kalema — APPROVED** [review]
Hi @Mahdi-Niazi 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# Congratulations! 🎉 


## Highlights

Tests are passing 👍 
PR is well descriptive 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@joel-kalema** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**